### DISCUSSION TITLE
Supporting outscale CAPI clusters

### DISCUSSION THREAD

**Discussion by pierreozoux** [clastix/cluster-api-control-plane-provider-kamaji#82] [Ideas]
Hi!

Really interested in your project.
How does it work to add another CAPI provider?
What is required? I could spend time trying it!

**Comment by prometherion** [comment]
Hey @pierreozoux, thanks for the shout-out, really appreciated!

Kamaji was born mostly to solve a specific problem, such as running Kubernetes Control Planes at scale leveraging on multi-tenancy. By interacting with the community we started evaluating Cluster API since the concept of orchestrating the cluster directly from Kubernetes could fit with running the CP in Kubernetes as a Pod. Starting from there, we collected some feedback from the community and the first implementation requested has been the OpenStack one.

Although Kamaji is totally open source and on its path to getting donated to a foundation, we're trying to follow what the community (when contributing as an individual, otherwise, customers when wearing the CLASTIX hat, which is the commercial company who developed Kamaji) is asking for.

If you have any specific infrastructure requirements we can start the discussion here, collecting the findings, and the requirements. Most of the time, the implementation is pretty straightforward as happened with the vSphere one (#9), sometimes a broader consensus is required with the CAPI infra provider community (as happening with the AWS one). Contributions are welcome, as the Nutanix one (#72), in case of limited programming skills, a demo environment offered by the requested is absolutely valuable and appreciated since we would be able to write code on our own and test it directly.

I just moved the discussion in this section @pierreozoux since we'd like to keep the _Issue_ section only for the bug reports, hope you can understand it.

**Comment by prometherion** [comment]
A small update: starting from Cluster API v1.8 the Control Plane implementation can provide the Control Plane endpoint, pretty similar to what is available for any Infrastructure provider.

With #137 any newer Cluster API provider adhering to the v1.8 contract can extract this information, and use it to continue the process of the cluster creation.

One of my tasks is to engage with the currently supported infrastructure provider maintainers and get a confirmation if upcoming versions will support this contract, once we have an overall agree we'll withdraw the Infrastructure Cluster patching to have a smooth experience!
